src/wp-admin/options-privacy.php

```diff
@@ -103,7 +103,7 @@ static function ( $body_class ) {
 				'error'
 			);
 		} else {
-			update_option( 'wp_page_for_privacy_policy', $privacy_policy_page_id );
+			update_option( 'wp_page_for_privacy_policy', $privacy_policy_page_id, true );
```

Comment:
This still needs to be reverted per https://github.com/WordPress/wordpress-develop/pull/6465#discussion_r1731897669.
```suggestion
			update_option( 'wp_page_for_privacy_policy', $privacy_policy_page_id );
```